dlt/dlt_broker_handlers.py

```diff
@@ -306,6 +295,117 @@ def handle(self, message):
+                    # So, no unittest could be done for this error handling
+                    if retries_for_non_existing_file == 0:
+                        logger.error("After retries, dlt file %s still does not exist", self.file_name)
+                        raise err
+                    logger.warning(
+                        "DLT file %s does not exist, will try %d times again",
+                        self.file_name,
+                        retries_for_non_existing_file,
+                    )
+                    retries_for_non_existing_file = retries_for_non_existing_file - 1
+                    time.sleep(1)
+                else:
+                    raise err
+            except Exception:  # pylint: disable=broad-except
+                logger.exception("Exception during the DLT message receive")
+
+        logger.debug("DLTFileSpinner starts to quit...")
+        if self.dlt_reader.stop_reading_proc.is_set():
+            self.dlt_reader.stop_reading_proc.set()
+        self.message_queue.close()
+        logger.info("DLTFileSpinner worker execution complete")
+
+    def stop(self):
+        self.dlt_reader.stop_reading_proc.set()
+        self.mp_stop_flag.set()
+
+
+class DLTMessageHandler(DLTMessageDispatcherBase):
+    """Process receiving the DLT messages and handing them to DLTContextHandler
+
+    This process instance is responsible for collecting messages from
+    the DLT daemon, tagging them with the correct queue id and placing
+    them on the messages queue.
+    """
+
+    def __init__(
+        self, filter_queue, message_queue, mp_stop_event, client_cfg, dlt_time_value=None, filter_ack_queue=None
+    ):
+        super().__init__(filter_queue, message_queue, mp_stop_event, dlt_time_value, filter_ack_queue)
+        self._ip_address = client_cfg["ip_address"]
+        self._port = client_cfg.get("port", DLT_DAEMON_TCP_PORT)
+        self._filename = client_cfg.get("filename")
+        self.verbose = client_cfg.get("verbose", 0)
+        self.timeout = client_cfg.get("timeout", DLT_CLIENT_TIMEOUT)
+        self._client = None
+        self.tracefile = None
+
+    def is_valid_message(self, message):
+        return message and (message.apid != "" or message.ctid != "")
+
+    def _client_connect(self):
+        """Create a new DLTClient
+
+        :param int timeout: Time in seconds to wait for connection.
+        :returns: True if connected, False otherwise
+        :rtype: bool
+        """
+        logger.debug(
+            "Creating DLTClient (ip_address='%s', Port='%s', logfile='%s')",
+            self._ip_address,
+            self._port,
+            self._filename,
+        )
+        self._client = DLTClient(servIP=self._ip_address, port=self._port, verbose=self.verbose)
+        connected = self._client.connect(self.timeout)
+        if connected:
+            logger.info("DLTClient connected to %s", self._client.servIP)
+        return connected
+
```

Comment:
why not add

```
def stop():
    pass
```

then eliminate the if-block in DLTBroker stop function?

Reply:
already replied in private chat, and prefer to stay it as it is